dsa/src/generate/secret_number.rs

```diff
@@ -2,32 +2,16 @@
 //! Generate a per-message secret number
 //!
 
-use crate::{signing_key::SigningKey, Components};
-use alloc::{vec, vec::Vec};
+use crate::{Components, signing_key::SigningKey};
+use alloc::vec;
 use core::cmp::min;
-use digest::{core_api::BlockSizeUser, Digest, FixedOutputReset};
-use num_bigint::{BigUint, ModInverse, RandBigInt};
-use num_traits::{One, Zero};
-use rfc6979::HmacDrbg;
-use signature::rand_core::CryptoRngCore;
-use zeroize::Zeroize;
+use crypto_bigint::{BoxedUint, NonZero, RandomBits};
+use digest::{Digest, FixedOutputReset, core_api::BlockSizeUser};
+use signature::rand_core::TryCryptoRng;
+use zeroize::Zeroizing;
 
-/// Reduce the hash into an RFC-6979 appropriate form
-fn reduce_hash(q: &BigUint, hash: &[u8]) -> Vec<u8> {
-    // Reduce the hash modulo Q
-    let q_byte_len = q.bits() / 8;
-
-    let hash_len = min(hash.len(), q_byte_len);
-    let hash = &hash[..hash_len];
-
-    let hash = BigUint::from_bytes_be(hash);
-    let mut reduced = (hash % q).to_bytes_be();
-
-    while reduced.len() < q_byte_len {
-        reduced.insert(0, 0);
-    }
-
-    reduced
+fn reduce_hash(hash: &[u8], desired_size: usize) -> &[u8] {
+    &hash[(hash.len() - desired_size)..]
 }
```

Comment:
What happened here?

Reply:
The logic moved in `secret_number_rfc6979`, the name was incorrect, that should have been `truncate_hash` or something.